CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/rest/RestClientNGTest.java

```diff
@@ -0,0 +1,412 @@
+        System.out.println("testPostWithJson");
+        String url = "";
+        List<Tuple<String, String>> params = new ArrayList<>();
+        params.add(new Tuple<>("AAA", "BBB"));
+        params.add(new Tuple<>(null, "CCC"));
+        params.add(new Tuple<>(" ", "DDD"));
+        String json = "{\"jsonText\":\"I am JSON\"}";
+        Map<String, List<String>> headerFields = new HashMap<>();
+        headerFields.put("hdr1", new ArrayList<>());
+        RestClientImpl instance = new RestClientImpl(501, "message", headerFields);
+        outputStreamString = new StringBuilder();
+        instance.postWithJson(url, params, json);
+        assertEquals(instance.getResponseCode(), 501);
+        assertEquals(instance.getResponseMessage(), "message");
+        assertEquals(instance.getHeaderFields().size(), headerFields.size());
+        assertTrue(instance.getHeaderFields().containsKey("hdr1"));
+        assertNull(instance.getBytes());
+        assertEquals(outputStreamString.toString(), json);
+    }
+
+    /**
+     * Test of postWithBytes method, of class RestClient.
+     */
+    @Test
+    public void testPostWithBytes_String_Map() throws Exception {
+        System.out.println("testPostWithBytes_String_Map - not testing deprecated function");
+    }
+
+    /**
+     * Test of postWithBytes method, of class RestClient.
+     */
+    @Test
+    public void testPostWithBytes() throws Exception {
+        System.out.println("testPostWithBytes");
+        String url = "";
+        List<Tuple<String, String>> params = new ArrayList<>();
+        params.add(new Tuple<>("AAA", "BBB"));
+        params.add(new Tuple<>(null, "CCC"));
+        params.add(new Tuple<>(" ", "DDD"));
+        String byteString = "Sample Bytes From String";
+        byte[] bytes = byteString.getBytes();
+        Map<String, List<String>> headerFields = new HashMap<>();
+        headerFields.put("hdr1", new ArrayList<>());
+        RestClientImpl instance = new RestClientImpl(501, "message", headerFields);
+        outputStreamString = new StringBuilder();
+        instance.postWithBytes(url, params, bytes);
+        assertEquals(instance.getResponseCode(), 501);
+        assertEquals(instance.getResponseMessage(), "message");
+        assertEquals(instance.getHeaderFields().size(), headerFields.size());
+        assertTrue(instance.getHeaderFields().containsKey("hdr1"));
+        assertNull(instance.getBytes());
+        assertEquals(outputStreamString.toString(), byteString);
+    }
+
+    /**
+     * Test of afterPost method, of class RestClient.
+     */
+    @Test
+    public void testAfterPost() {
+        System.out.println("testAfterPost");
+        String url = "";
+        List<Tuple<String, String>> params = null;
+        RestClient instance = new RestClientImpl(0, "", null);
+        instance.afterPost(url, params);
+        // Method does nothing - designed to be overridden, test really just
+        // checks it can be called
```

Comment:
similar

Reply:
left in as they do add coverage, although as mentioned they don't test 'muxh'